lmfdb/number_fields/web_number_field.py

```diff
@@ -271,12 +271,17 @@ def formatfield(coef, show_poly=False, missing_text=None, data=None):
     if isinstance(coef, str):
         coef = string2list(coef)
     if data is None:
-        thefield = WebNumberField.from_coeffs(coef)
+        from lmfdb.number_fields.number_field import poly_to_field_label
+        pol = coeff_to_poly(coef)
+        label = poly_to_field_label(pol)
+        thefield = WebNumberField(label)
```

Comment:
OK, the delay is happening because of this change I made to `formatfield` in `web_number_field.py`

I thought this change was necessary for the following reason: previously, the kerpoly was stored in `char_dir_orbits`, it was looked up in the number field database, it was found, and all was well. 

In the new schema, with kerpoly being computed on the fly, the coefficients defining the number field might not be exactly the same as those found in the number field database, and I was finding that, with `formatfield` as previously implemented, it was not finding the number field from a possibly different polynomial (that actually defines the same number field).

So I decided to add these four lines, which essentially get the field label from the kerpoly. And getting this field label is precisely the step that takes time (about 30s in the `199.k` example).

So the question is: how to do this without taking this long? I wouldn't have thought finding the label of a number field from a polynomial would take that long ...

Reply:
I started to write exactly what @jwj61 wrote below: we cannot afford to call polredabs here unless the degree is quite small.

Note that this situation is different from when a user types a polynomial into the number field jump box, there the user presumably is willing to wait a bit to find the number field they care about and if it takes a few seconds (or even times out) that's okay, because the user asked for this.  But whatever we do here needs to be fast (e.g. 100 milliseconds or less).  Most of the time a user goes to a Dirichlet character page they are not going there to find the LMFDB label of the kernel field and we can't impose a performance penalty for computing it on the fly.

If we think it's important to have the LMFDB labels of the kernel fields, then perhaps we should precompute them up to some modulus and degree bound (where the modulus bound might be much less than the range of the Dirichlet character database) and store them in a separate table.  I also think it would be okay to polredabs the kernel field when the degree is quite small (maybe 12 or less?).  It might be worth asking @alexjbest what he thinks, I believe he is the one who added the kernel fields to the Dirichlet character pages.

As a side note, once this PR is merged I plan to push the modulus bound up to 1,000,000, so this is the scale we should be thinking about.

I'm going to resolve this conversation but would encourage others with opinions/suggestions to post them below.

Reply:
Just saw this reply ... only calling polredabs if the degree is <= 12 seems to be just as fast as prod on a few examples i checked. I still think the language needs to be changed to make clear that, when we're not calling polredabs on the kerpoly, then the number field *might* be in the database, and the user can check this via the number field jump.

For now I'm going to go ahead and commit these changes to this PR: revert `formatfield` back to its original, and call polredabs on the kerpoly if the degree <= 12.